qiskit_machine_learning/neural_networks/two_layer_qnn.py

```diff
@@ -63,37 +64,49 @@ def __init__(
         # check num_qubits, feature_map, and ansatz
         if num_qubits is None and feature_map is None and ansatz is None:
             raise QiskitMachineLearningError(
-                "Need at least one of num_qubits, feature_map, or ansatz!"
+                "Need at least one of num_qubits, feature_map, or ansatz but all are None!"
             )
         num_qubits_: int = None
         feature_map_: QuantumCircuit = None
         ansatz_: QuantumCircuit = None
-        if num_qubits:
+        if num_qubits is not None:
             num_qubits_ = num_qubits
-            if feature_map:
+            if feature_map is not None:
                 if feature_map.num_qubits != num_qubits:
-                    raise QiskitMachineLearningError("Incompatible num_qubits and feature_map!")
+                    raise QiskitMachineLearningError(
+                        f"Mismatching num_qubits ({num_qubits}) and number of qubits "
+                        f"in the feature_map ({feature_map.num_qubits})!"
+                    )
```

Comment:
For the feature map and anstaz, if these are blueprint ccts we could set the num qubits similar to what is done in VQE right where of course we cannot do this for plain ccts. Just wondering if perhaps it could be more flexible in this regard.

Reply:
Yeah I think we could!

Reply:
@woodsp-ibm Can you please point me out to the code in VQE you are referring to? I don't fully understand where the problem is. Otherwise, I fine with the PR.

Reply:
@adekusar-drl  There is not a "problem" per-se more I suggesting, like we do in VQE, that potentially here we could adjust the circuit to the number of qubits where possible rather than simply throwing an exception. If you know the number of qubits then one could argue, that if you build out a circuit, which is a blueprint circuit, that you as a caller can also set its number of qubits so it's matched. Since this was often not known in VQE, when the constructor was given an ansatz circuit, we set it later based on the operator, where possible, to match. https://github.com/Qiskit/qiskit-terra/blob/573198bdc55eaf4ad0691d441ae41c874f46313d/qiskit/algorithms/minimum_eigen_solvers/vqe.py#L308-L320

Reply:
Oh I see now. Yes, this can be done as in VQE, but I suggest to move to a separate PR and merge this one. In VQC we can introduce the same improvement. I can take care of your proposal.